internal/envoy/config.go

```diff
@@ -76,6 +76,18 @@ static_resources:
         port_value: {{ if .XDSGRPCPort }}{{ .XDSGRPCPort }}{{ else }}8001{{ end }}
     lb_policy: ROUND_ROBIN
     http2_protocol_options: {}
+    circuit_breakers:
+      thresholds:
+        - priority: high
+          max_connections: 100000
+          max_pending_requests: 100000
+          max_requests: 60000000
+          max_retries: 50
```

Comment:
I'm not sure about this value. What I see in testing is 
```
contour::127.0.0.1:8001::cx_active::1
contour::127.0.0.1:8001::cx_connect_fail::2
contour::127.0.0.1:8001::cx_total::3
contour::127.0.0.1:8001::rq_active::514
contour::127.0.0.1:8001::rq_error::4
contour::127.0.0.1:8001::rq_success::0
contour::127.0.0.1:8001::rq_timeout::0
contour::127.0.0.1:8001::rq_total::518
contour::127.0.0.1:8001::health_flags::healthy
contour::127.0.0.1:8001::weight::1
contour::127.0.0.1:8001::region::
contour::127.0.0.1:8001::zone::
contour::127.0.0.1:8001::sub_zone::
contour::127.0.0.1:8001::canary::false
contour::127.0.0.1:8001::success_rate::-1
```
518 in progress connections, but nothing has timed out yet. Retrying is fine, even retrying indefinitely is fine, but I think a reasonably short _connection_ timeout, say < 5 seconds, should be applied to each attempt.

It doesn't matter so much for this change as we're going to embiggen the max incoming connections on the server side in #308, but we should revisit it in #313